src/ui.h

```diff
@@ -193,16 +207,25 @@ class uimenu: public ui_container
 
         uimenu_callback *callback;
 
+        // legacy constructors
         uimenu( const std::string &hotkeys = "" ); // bare init
 
-        uimenu( bool cancancel, const char *message, ... ); // legacy menu()
         uimenu( bool cancelable, const char *mes,
                 const std::vector<std::string> options ); // legacy menu_vec
-        uimenu( bool cancelable, const char *mes, const std::vector<std::string> &options,
-                const std::string &hotkeys );
         uimenu( bool cancelable, int startx, int width, int starty, std::string title,
                 std::vector<uimenu_entry> ents );
-        uimenu( int startx, int width, int starty, std::string title, std::vector<uimenu_entry> ents );
+
+        // migration in process
+        uimenu( uimenu_migration_tag );
```

Comment:
I have to add this migration tag so I can migrate the default constructions once at a time. In the final PR I'll remove this tag. I can't think of any better way to do this.